ts/src/main/scala/ch/epfl/ts/brokers/ExampleBroker.scala

```diff
@@ -0,0 +1,39 @@
+package ch.epfl.ts.brokers
+
+import ch.epfl.ts.component.Component
+import ch.epfl.ts.data.{Order, ConfirmRegistration, Register}
+import akka.actor.{ActorLogging, Props, ActorRef}
+import ch.epfl.ts.engine.{GetWalletFunds, Wallet}
+
+/**
+ * Created by sygi on 03.04.15.
+ */
+class ExampleBroker extends Component with ActorLogging { //TODO(sygi): println -> log.debug
+  var mapping = Map[Long, ActorRef]()
+  override def receiver: PartialFunction[Any, Unit] = {
+    case Register(id) => {
+      println("Broker: registration of agent " + id)
+      println("with ref: " + sender())
+      if (mapping.get(id) != None){
+        println("Duplicate Id")
+        //TODO(sygi): reply to the trader that registration failed
+        //TODO(sygi): return?
+      }
+      mapping = mapping + (id -> sender())
+      context.actorOf(Props[Wallet], "wallet" + id)
+      sender() ! ConfirmRegistration
+    }
+    case GetWalletFunds(uid) => { //TODO(sygi): check if trader asks for his wallet
```

Comment:
Haha yes or otherwise this leaves the door open to some very interesting strategies involving other trader's wallets :hamburger: